numpy/lib/format.py

```diff
@@ -313,6 +313,9 @@ def descr_to_dtype(descr):
 
         offset += dt.itemsize
 
+    if not fields:
+        return numpy.dtype([])
```

Comment:
This doesn't respect the itemsize.

I'd recommend fixing this by replacing `fields.append((name, dt, offset))` with `names.append(name)`, `formats.append(dt)`, etc

Reply:
I tried the following
```
In [7]: np.dtype({'names': [], 'formats': [], 'titles': [], 'offsets': [], 'itemsize': 0})                                                                                                            
Out[7]: dtype([])

In [8]: np.dtype({'names': [], 'formats': [], 'titles': [], 'offsets': [], 'itemsize': 0})  == np.dtype([])                                                                                                   
Out[8]: True
```
So there is no difference.

Reply:
`itemsize` might not be 0.

Reply:
What exactly is the itemsize for an empty array with empty descr?

Reply:
It depends on which empty descriptor. Your example has size 0, but you can construct one with any size you like using the dictionary form.

This code contains a loop that does just that.